api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/WeightLoader.kt

```diff
@@ -183,108 +86,49 @@ public fun GraphTrainableModel.loadWeightsByPathTemplates(
     hdfFile: HdfFile,
     kernelDataPathTemplate: String = KERNEL_DATA_PATH_TEMPLATE, // TODO: doesnt' work for batchnorm/depthwise
     biasDataPathTemplate: String = BIAS_DATA_PATH_TEMPLATE
-) {
-    check(this.isModelCompiled) { "The model is not compiled yet. Compile the model to use this method." }
-    check(!isModelInitialized) { "Model is initialized already!" }
-    this.logger.info { "Starting weights loading.." }
-    this.layers.forEach {
-        run {
-            fillLayerWeights(
-                it,
-                hdfFile,
-                LayerConvOrDensePaths("", kernelDataPathTemplate, biasDataPathTemplate),
-                this
-            ) // TODO: doesnt' work for batchnorm/depthwise
-        }
-    }
-    this.logger.info { "Weights are loaded." }
-    this.isModelInitialized = true
-}
+): Unit = loadWeightsByPathTemplates(hdfFile, layers, kernelDataPathTemplate, biasDataPathTemplate)
 
 /**
- * Loads weights from hdf5 file created in Keras TensorFlow framework for pre-defined list of layers.
+ * Loads weights from hdf5 file created in Keras TensorFlow framework for non-trainable (or frozen) layers only.
  *
- * NOTE: Weights for another layers will not be loaded (should be initialized manually).
+ * NOTE: Weights for trainable layers will not be loaded and will be initialized via default initializers.
  *
  * @param [hdfFile] File in hdf5 file format containing weights of Sequential model.
- * @param [layerList] List of layers to load weights. Weights for other layers will be initialized by initializer later.
  * @param [kernelDataPathTemplate] Template path to kernel weights of the specific layer.
  * @param [biasDataPathTemplate] Template path to bias weights of the specific layer.
  */
-public fun GraphTrainableModel.loadWeightsByPathTemplates(
+public fun Functional.loadWeightsForFrozenLayersByPathTemplates(
```

Comment:
@juliabeliaeva could you please explain excluding the Sequential from the ability to have this function, I suppose it could the common functionality (agree, that there is no examples in the codebase)

Reply:
It was like this in `master`, I did not change it. See https://github.com/JetBrains/KotlinDL/blob/11d4eb175323b61d2e709d4133cbbde49c033746/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/WeightLoader.kt#L278

Reply:
Changed the type to `GraphTrainableModel`.

Reply:
Great!